Fixed "exception in selector loop" on Linux

Helias opened the pull request:
Fixed unexpected exception in the selector loop on Linux by upgrading netty in gradle settings.
I upgraded netty to 4.1.5.Final version to fix the follow error on Linux.

```
WARN  [io.netty.channel.epoll.EpollEventLoop] - Unexpected exception in the selector loop.
java.lang.NoSuchMethodError: io.netty.channel.epoll.EpollEventLoop.deadlineNanos()J
  at io.netty.channel.epoll.EpollEventLoop.epollWait(EpollEventLoop.java:243)
  at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:278)
  at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
  at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
  at java.base/java.lang.Thread.run(Thread.java:844)
```

emansom commented:
P.S: I've been playing around with [gdbgui](https://gdbgui.com/), pretty cool debugging tool. Helps when debugging segmentation faults. :smile: 

P.P.S; there's a race condition in the roller_task, sometimes it tries to move a user, while that user is currently moving to another room, resulting in accessing a null pointer.

emansom commented:
I'm now forcing SQLite to use [serialized mode](https://www.sqlite.org/threadsafe.html), if I understand the documentation of SQLite correctly; this should for now resolve the contention issues.

emansom commented:
As per [the docs](https://www.sqlite.org/c3ref/prepare.html) advisement `The preferred routine to use is sqlite3_prepare_v2(). The sqlite3_prepare() interface is legacy and should be avoided.`

emansom commented:
`database is locked` errors might still happen with WAL and serialized, see this [post](https://stackoverflow.com/a/17124940). In order to solve this completely we need to wrap a lock/mutex around the sqlite3 functions inside the query functions. Then there's only one connection open at a time.

emansom commented:
Switching to multi-thread mode as the serialized mode is safer, but blows up the callee with `SQL_BUSY` errors.

As per the [wiki](https://www.sqlite.org/threadsafe.html) `Multi-thread. In this mode, SQLite can be safely used by multiple threads provided that no single database connection is used simultaneously in two or more threads.` Note **single**. We do not pass around database connections between threads, therefore should be safe.

I hope this solves the `database is locked` errors. Testing will tell. :smile:

Quackster commented:
Hmm, I think long term will be better to use PostgreSQL, otherwise the source file with all these SQLite queries will just continue to grow and grow until the amount of work required to convert over to PostgreSQL becomes exponentially huge to the point where it was regrettable to not do it sooner.

Quackster commented:
Also this may fix the race condition when the roller task attempts to slide a user who is changing rooms: https://github.com/Quackster/Kepler/commit/e1aec1c11cd3a34ddd9e3a93b1592d4eb73b9722

emansom commented:
I'd prefer MySQL/MariaDB, as the community has more knowledge of that DBMS. There has to be better bindings out there.

Helias committed:
Fixed "exception in selector loop" on Linux

Fixed unexpected exception in the selector loop on Linux by upgrading netty in gradle settings.

Quackster commented:
Thanks. 😄

emansom commented:
What the actual f is up with Netty
